src/fund/components/FundCardAmountInput.tsx

```diff
@@ -0,0 +1,197 @@
+
+  const inputRef = useRef<HTMLInputElement>(null);
+  const hiddenSpanRef = useRef<HTMLSpanElement>(null);
+  const currencySpanRef = useRef<HTMLSpanElement>(null);
+
+  const value = inputType === 'fiat' ? fiatValue : cryptoValue;
+
+  const handleChange = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      let value = e.target.value;
+
+      value = formatDecimalInputValue(value);
+
+      if (inputType === 'fiat') {
+        const fiatValue = limitToDecimalPlaces(value, 2);
+        setFiatValue(fiatValue);
+
+        // Calculate the crypto value based on the exchange rate
+        const calculatedCryptoValue = String(
+          Number(value) * Number(exchangeRate),
+        );
+
+        const resultCryptoValue = limitToDecimalPlaces(
+          calculatedCryptoValue,
+          8,
+        );
+        setCryptoValue(calculatedCryptoValue === '0' ? '' : resultCryptoValue);
+      } else {
+        setCryptoValue(value);
+
+        // Calculate the fiat value based on the exchange rate
+        const calculatedFiatValue = String(
+          Number(value) / Number(exchangeRate),
+        );
+        const resultFiatValue = limitToDecimalPlaces(calculatedFiatValue, 2);
+        setFiatValue(resultFiatValue === '0' ? '' : resultFiatValue);
+      }
+    },
+    [exchangeRate, setFiatValue, setCryptoValue, inputType],
+  );
+
+  // biome-ignore lint/correctness/useExhaustiveDependencies: When value changes, we want to update the input width
+  useEffect(() => {
+    if (hiddenSpanRef.current) {
+      const width = Math.max(42, hiddenSpanRef.current.offsetWidth);
+      const currencyWidth =
+        currencySpanRef.current?.getBoundingClientRect().width || 0;
+
+      // Set the input width based on the span width
+      if (inputRef.current) {
+        inputRef.current.style.width = `${width}px`;
+        inputRef.current.style.maxWidth = `${390 - currencyWidth}px`;
+      }
+    }
+  }, [value]);
+
+  // biome-ignore lint/correctness/useExhaustiveDependencies: We only want to focus the input when the input type changes
+  useEffect(() => {
+    // focus the input when the input type changes
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [inputType]);
+
+  return (
+    <div className="flex py-6">
+      <style>
+        {`
+          input[type="number"]::-webkit-inner-spin-button,
+          input[type="number"]::-webkit-outer-spin-button {
+            -webkit-appearance: none;
+            appearance: none;
+            margin: 0;
+          }
+
+          input[type="number"] {
+            -moz-appearance: textfield;
+          }
+        `}
+      </style>
+
+      <div className="flex" style={{ height: '78px' }}>
+        {/* Display the fiat currency sign before the input*/}
+        {inputType === 'fiat' && currencySign && (
+          <FundCardCurrencyLabel
+            ref={currencySpanRef}
+            currencySign={currencySign}
+          />
+        )}
+
+        <input
```

Comment:
It would be nice if the input spanned the entire width, otherwise if a user clicks to the right, they won't focus the input
![CleanShot 2024-12-18 at 13 31 46@2x](https://github.com/user-attachments/assets/7d33467a-57b3-470d-9be7-7be4b1685f74)

Reply:
Unfortunately currently it is working this way because if you switch to Crypto it will have a space [1 - - - - - - -ETH] but we want the asset name next to the digits [1 ETH].  
But you are right, we should make it better for users. I'll take a look into focusing on the input field when that area is clicked. (I still have a few features to add to this component, so I'll make this change in a follow up PR )